newa/__init__.py

```diff
@@ -382,12 +399,31 @@ def from_errata_tool(cls, event: Event) -> list[Erratum]:
 
 @define
 class Issue(Cloneable, Serializable):
-    """ A Jira issue """
+    """
+    A Jira issue
 
-    id: str
+    Issue is identified by its key in Jira (eg. NEWA-123).
+    We don't associate the details of the issue, i.e. content of Jira issue,
+    by default but only on demand because it requires data retrieval from
+    Jira and it is not always needed.
+    """
 
-    def fetch_details(self) -> None:
-        raise NotImplementedError
+    id: str = field()
+    _details: JiraIssue = field(repr=False,
```

Comment:
Must be `Optional[JiraIssue]` because the default is `None`.

Reply:
Fixed.